app/src/main/java/cl/maleb/testcarsales/ui/covid/CovidViewModel.kt

```diff
@@ -0,0 +1,71 @@
+package cl.maleb.testcarsales.ui.covid
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import cl.maleb.testcarsales.data.Covid
+import cl.maleb.testcarsales.data.CovidRepository
+import cl.maleb.testcarsales.data.Result
+import java.util.*
+import javax.inject.Inject
+import kotlin.properties.Delegates
+
+class CovidViewModel @Inject constructor(private val covidRepository: CovidRepository) :
+    ViewModel() {
+
+    var Date by Delegates.notNull<String>()
+    var isShowProgress = MutableLiveData<Int>().apply { value = 0 }
+    var isShowContent = MutableLiveData<Int>().apply { value = 0 }
+
+    val covid by lazy { covidRepository.getCovidData(Date) }
+
+    fun loadCovid(): LiveData<Result<Covid>> {
+        return covidRepository.getCovidData(Date)
+    }
+
+    fun getCurrentDay(): String {
+        val c = Calendar.getInstance()
+        val year = c.get(Calendar.YEAR)
```

Comment:
I would create an extension and not doing this work here

Also, please, use DateFormatter! No hardcoded string is allowed